src/Table/TableBody.js

```diff
@@ -125,7 +125,11 @@ class TableBody extends Component {
   };
 
   componentWillReceiveProps(nextProps) {
-    if (this.props.allRowsSelected && !nextProps.allRowsSelected) {
+    if (!this.props.allRowsSelected && nextProps.allRowsSelected) {
```

Comment:
This is to clear out the selected state within the component when the `allRowsSelected` checkbox is checked. This allows for correct handling when it is unchecked or a single row is checked.